File: src/main/java/snake/difficulty/Intermediate.java
package snake.difficulty;

public class Intermediate extends Difficulty {
    public Intermediate() {
        this.delay = 150;
    }

    @Override
    public void changeDelay(int foodConsumed) {
        if (delay > 50 && foodConsumed % 5 == 0) {
            delay -= 15;
        }
    }
}

File: docs/difficulty/Intermediate.js
import {Difficulty} from "./Difficulty.js";
export class Intermediate extends Difficulty {
    constructor() {
        super(150);
    }
    changeDelay(foodConsumed) {
        if (this.delay > 50 && foodConsumed % 5 === 0) {
            this.delay -= 15;
        }
    }
}
